openpype/hosts/maya/plugins/publish/validate_gltf_textures_names.py

```diff
@@ -0,0 +1,129 @@
+            main_shader = cmds.listConnections(material,
+                                               destination=True,
+                                               type="StingrayPBS")
+            for shader in main_shader:
+                albedo = cmds.listConnections(shader + ".TEX_color_map")[0]
+                dif_path = cmds.getAttr(albedo + ".fileTextureName")
+                dif = dif_path.split(".")[0]
+                # "_D"
+                if not dif.endswith("_D"):
+                    invalid.add(dif_path)
+                orm_packed = cmds.listConnections(shader + ".TEX_ao_map")[0]
+                if orm_packed:
+                    # "_ORM"
+                    orm_path = cmds.getAttr(orm_packed + ".fileTextureName")
+                    orm = orm_path.split(".")[0]
+                    if not orm.endswith("_ORM"):
+                        invalid.add(orm_path)
+                nrm = cmds.listConnections(shader + ".TEX_normal_map")[0]
+                if nrm:
+                    nrm_path = cmds.getAttr(nrm + ".fileTextureName")
+                    nrm_map = nrm_path.split(".")[0]
+                    # "_N"
+                    if not nrm_map.endswith("_N"):
+                        invalid.add(nrm_path)
+
+        return list(invalid)
+
+    def get_texture_node_invalid(self, instance):
+        invalid = set()
+        shading_grp = self.get_material_from_shapes(instance)
+        for material in shading_grp:
+            main_shader = cmds.listConnections(material,
+                                               destination=True,
+                                               type="StingrayPBS")
+            for shader in main_shader:
+                # diffuse texture file node
+                albedo = cmds.listConnections(shader + ".TEX_color_map")
+                if not albedo:
+                    invalid.add(albedo)
+        return list(invalid)
+
+    def get_texture_shader_invalid(self, instance):
+
+        invalid = set()
+        shading_grp = self.get_material_from_shapes(instance)
+        for shading_group in shading_grp:
+            material_name = "{}.surfaceShader".format(shading_group)
+            material = cmds.listConnections(material_name,
+                                            source=True,
+                                            destination=False,
+                                            type="StingrayPBS")
+
+            if not material:
+                # add material name
+                material = cmds.listConnections(material_name)[0]
+                invalid.add(material)
+        return list(invalid)
+
+    def get_material_from_shapes(self, instance):
+        shapes = cmds.ls(instance, type="mesh", long=True)
+        for shape in shapes:
+            shading_grp = cmds.listConnections(shape,
+                                               destination=True,
+                                               type="shadingEngine")
+
+            return shading_grp
```

Comment:
wouldn't this return shading group for the first shape it founds?

Reply:
It would yes. It might have even return None if the first shape had no shading engine assigned.